ugsync/src/main/java/org/apache/ranger/ldapusersync/process/CustomSSLSocketFactory.java

```diff
@@ -30,89 +42,70 @@
-    	String keyStoreFile =  config.getSSLKeyStorePath();
-    	String keyStoreFilepwd = config.getSSLKeyStorePathPassword();
-    	String trustStoreFile = config.getSSLTrustStorePath();
-    	String trustStoreFilepwd = config.getSSLTrustStorePathPassword();
-    	String keyStoreType = config.getSSLKeyStoreType();
-    	String trustStoreType = config.getSSLTrustStoreType();
-    	try {
-
-			KeyManager[] kmList = null;
-			TrustManager[] tmList = null;
-
-			if (keyStoreFile != null && keyStoreFilepwd != null) {
-
-				KeyStore keyStore = KeyStore.getInstance(keyStoreType);
-				InputStream in = null;
-				try {
-					in = getFileInputStream(keyStoreFile);
-					if (in == null) {
-						LOG.error("Unable to obtain keystore from file [" + keyStoreFile + "]");
-						return;
-					}
-					keyStore.load(in, keyStoreFilepwd.toCharArray());
-					KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
-					keyManagerFactory.init(keyStore, keyStoreFilepwd.toCharArray());
-					kmList = keyManagerFactory.getKeyManagers();
-				}
-				finally {
-					if (in != null) {
-						in.close();
-					}
-				}
-				
-			}
-
-			if (trustStoreFile != null && trustStoreFilepwd != null) {
-
-				KeyStore trustStore = KeyStore.getInstance(trustStoreType);
-				InputStream in = null;
-				try {
-					in = getFileInputStream(trustStoreFile);
-					if (in == null) {
-						LOG.error("Unable to obtain keystore from file [" + trustStoreFile + "]");
-						return;
-					}
-					trustStore.load(in, trustStoreFilepwd.toCharArray());
-					TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
-					trustManagerFactory.init(trustStore);
-					tmList = trustManagerFactory.getTrustManagers();
-				}
-				finally {
-					if (in != null) {
-						in.close();
-					}
-				}
-			}
-
-			sslContext = SSLContext.getInstance("TLSv1.2");
-
-			sslContext.init(kmList, tmList, new SecureRandom());
-			sockFactory = sslContext.getSocketFactory();
-			}
-			catch(Throwable t) {
-				throw new RuntimeException("Unable to create SSLConext for communication to policy manager", t);
-			}
+        SSLContext sslContext        = null;
+        String     keyStoreFile      = config.getSSLKeyStorePath();
+        String     keyStoreFilepwd   = config.getSSLKeyStorePathPassword();
+        String     trustStoreFile    = config.getSSLTrustStorePath();
+        String     trustStoreFilepwd = config.getSSLTrustStorePathPassword();
+        String     keyStoreType      = config.getSSLKeyStoreType();
+        String     trustStoreType    = config.getSSLTrustStoreType();
+        try {
+            KeyManager[]   kmList = null;
+            TrustManager[] tmList = null;
+
+            if (keyStoreFile != null && keyStoreFilepwd != null) {
+                KeyStore    keyStore = KeyStore.getInstance(keyStoreType);
+                InputStream in       = null;
+                try {
```

Comment:
Consider replacing finally with  try-with-resource: `try (InputStream in getFileInputStream(keyStoreFile)) {`.

Same at line 85 as well.